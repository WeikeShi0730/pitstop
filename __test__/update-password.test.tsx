import { render, RenderResult } from "@testing-library/react";
import UpdatePassword from "../components/update-password.component";

let documentBody: RenderResult;
describe("<UpdatePassword />", () => {
  beforeEach(
    () => (documentBody = render(<UpdatePassword currentUser={null} />))
  );
  it("Test <UpdatePassword />", () => {
    expect(
      documentBody.getByRole("button", {
        name: /Send a password update email/i,
      })
    ).toBeInTheDocument();
    const { baseElement } = documentBody;
    expect(baseElement).toMatchSnapshot();
  });
});
