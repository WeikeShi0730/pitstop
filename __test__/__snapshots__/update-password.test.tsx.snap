// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpdatePassword /> Test <UpdatePassword /> 1`] = `
<body>
  <div>
    <div
      class="text-center self-start m-3 text-lg md:text-xl"
    >
      Update my password?
    </div>
    <button
      class="text-sm md:text-base m-3 bg-slate-700 py-2 px-4 text-slate-100 rounded outline-none font-light hover:shadow-md hover:shadow-slate-700 hover:bg-slate-500 hover:text-white transition-all ease-in-out duration-200"
    >
      Send a password update email
    </button>
  </div>
</body>
`;
